components/theme/ThemedText.tsx

```diff
@@ -1,61 +1,47 @@
-import { Text, type TextProps, StyleSheet } from 'react-native';
+import { Text, type TextProps, StyleSheet } from "react-native";
 
-import { useThemeColor } from '@/hooks/useThemeColor';
+import { useThemeColor } from "@/hooks/useThemeColor";
+import { Colors } from "@/constants/Colors";
+import { TextStyles } from "@/constants/Text";
 
 export type ThemedTextProps = TextProps & {
   lightColor?: string;
   darkColor?: string;
-  type?: 'default' | 'title' | 'defaultSemiBold' | 'subtitle' | 'link' | 'small' | 'smallSemiBold';
+  colorType?: keyof typeof Colors.light & keyof typeof Colors.dark;
+  type?: keyof typeof TextStyles;
 };
 
 export function ThemedText({
   style,
   lightColor,
   darkColor,
-  type = 'default',
+  type = "default",
+  colorType,
   ...rest
 }: ThemedTextProps) {
-  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
+  const color = useThemeColor(
+    { light: lightColor, dark: darkColor },
+    colorType ?? "textPrimary",
+  );
 
   return (
     <Text
       style={[
         { color },
-        type === 'default' ? styles.default : undefined,
-        type === 'title' ? styles.title : undefined,
-        type === 'defaultSemiBold' ? styles.defaultSemiBold : undefined,
-        type === 'subtitle' ? styles.subtitle : undefined,
-        type === 'link' ? styles.link : undefined,
-        type === 'small' ? { fontSize: 10 } : undefined,
-        type === 'smallSemiBold' ? { fontSize: 12, fontWeight: '600' } : undefined,
+        type === "superTitle" ? TextStyles.superTitle : undefined,
```

Comment:
I believe this whole section could be replaced by a `TextStyles[type]`. I'm not sure this would provide the same effect however, so TDB.

Reply:
Absolutely ! Thanks for catching that, as all styles weren't inside a stylesheet at the beginning I didn't think of it !